### ISSUE TITLE
[🐛 Bug]: Unable to switch back to default window

### ISSUE THREAD

**Issue by MihirModiAC** [SeleniumHQ/selenium#10142]
### What happened?

I upgraded to selenium 4.1.0 binding. When executing a test case which has steps:
To Switch to iframe 
Enter some value
Switch back to defaultWindow 
Click a button

Passes on local machine but when executing on selenium grid 4.1.0, it fails with error "Error communicating with the remote browser. It may have died."

Test case fails at "Switch back to defaultWindow" step.

### How can we reproduce the issue?

```shell
Go to site with Iframe
Switch to iframe 
Enter some value (e.g. work in iframe)
Switch back to defaultWindow (This is the step test case fails)
Click a button
```

### Relevant log output

```shell
org.openqa.selenium.remote.UnreachableBrowserException: Error communicating with the remote browser. It may have died.
Build info: version: '4.1.0', revision: '87802e897b'
System info: host: 'MMODI-CHI', ip: '192.168.1.62', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.16', java.version: '11.0.5'
Driver info: org.openqa.selenium.remote.RemoteWebDriver
Command: [05340cc11c0d83ffe3bd398767a23dc4af5601a0, switchToFrame {id=null}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 96.0.4664.45, chrome: {chromedriverVersion: 96.0.4664.35 (8df3babf8da8f..., userDataDir: /var/folders/yj/h3bj394n2tn...}, goog:chromeOptions: {debuggerAddress: localhost:53048}, javascriptEnabled: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: MAC, platformName: MAC, proxy: Proxy(), se:cdpVersion: 96.0.4664.45, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Session ID: 05340cc11c0d83ffe3bd398767a23dc4af5601a0
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:583)
	at org.openqa.selenium.remote.RemoteWebDriver$RemoteTargetLocator.defaultContent(RemoteWebDriver.java:1042)
```

### Operating System

Mac BigSur

### Selenium version

4.1.0

### What are the browser(s) and version(s) where you see this issue?

Chrome 96

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver 96.0.4664.45

### Are you using Selenium Grid?

4.1.0

**Comment by praveendvd** [comment]
```
describe('My Login application', () => {
    it('should login with valid credentials', async () => {
        
        await browser.url('https://selectorshub.com/xpath-practice-page/');
        await expect( $('h3=Food Time')).toBeExisting();
        await browser.switchToFrame(await $('#pact'))
        await expect( $('h3=Food Time')).not.toBeExisting();
        await expect( $('#jest')).toBeExisting();
        await browser.switchToParentFrame();
        await expect($('#jest')).not.toBeExisting();
    });
});
```

It works as expected

**Comment by MihirModiAC** [comment]
The defect is in Selenium Grid 4.1.0 not local execution.

I mentioned that local execution works but Selenium Grid has problem.

Also I am using the "switchToDefault" command not "switchToParent".

Thank you
--

Mihir Modi
ActiveCampaign / Staff Software Engineer
***@***.***
1 North Dearborn St, 5th Floor, Chicago IL, 60602, USA
<https://www.facebook.com/activecampaign>
<http://www.twitter.com/activecampaign>
<https://www.linkedin.com/company/activecampaign-inc->
<https://www.instagram.com/activecampaign/>

<https://www.activecampaign.com/sig/?u=mmodi&c=1>

**Comment by denysLystopadskyy** [comment]
Hello!

I have the same issue and deep down stack trace I've found:
```
Caused by:
                java.lang.NullPointerException
                    at org.openqa.selenium.remote.TracedCommandExecutor.execute(TracedCommandExecutor.java:47)
                    at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:559)
                    ... 19 more
```

so we put null here:
`execute(DriverCommand.SWITCH_TO_FRAME(null));`

and then receive NPE here:
`commandSpan.setAttribute("parameter." + parameter.getKey(), parameter.getValue().toString());`

**Comment by stevenerat** [comment]
**Env**:

Host: Mac 10.15.7
Remote Host: Windows 10
Selenium 4.1.0 in standalone grid mode
Chrome: 96.0.4664.110, chrome
ChromeDriver: 96.0.4664.45

**Reproducible case:**

```

    @Test
    public void testFrames() throws MalformedURLException {

        // Setup RemoteWebDriver
        ChromeOptions options = new ChromeOptions();
        options.setPlatformName(Platform.ANY.toString());
        String hub = "http://<HUB_IP>:4444";
        driver = new RemoteWebDriver(new URL(hub), options);
        ((RemoteWebDriver) driver).setFileDetector(new LocalFileDetector());

        try {
            // Example iFrame site
            driver.get("https://codepen.io/wendee/pen/kGaqs");

            // Switch to iFrame
            driver.switchTo().frame("result");

            // Return to default content
            driver.switchTo().defaultContent();
        }
        finally {
            driver.quit();
        }
    }
```

**Issue:**

```


Dec 14, 2021 1:55:04 PM org.openqa.selenium.remote.tracing.opentelemetry.OpenTelemetryTracer createTracer
INFO: Using OpenTelemetry for tracing
Dec 14, 2021 1:55:15 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C

org.openqa.selenium.remote.UnreachableBrowserException: Error communicating with the remote browser. It may have died.
Build info: version: '4.1.0', revision: '87802e897b'
System info: host: 'XXX', ip: 'fd4f:a118:adb5:2c:dc54:626d:eed3:5505%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.15.7', java.version: '1.8.0_275'
Driver info: org.openqa.selenium.remote.RemoteWebDriver
Command: [4ba2418b52fe6f1127be73aebc0a277f, switchToFrame {id=null}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 96.0.4664.110, chrome: {chromedriverVersion: 96.0.4664.45 (76e4c1bb2ab46..., userDataDir: C:\Users\XXXX\AppData\Lo...}, goog:chromeOptions: {debuggerAddress: localhost:60908}, javascriptEnabled: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: WINDOWS, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://<HUB_IP>:4444/ses..., se:cdpVersion: 96.0.4664.110, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Session ID: XXXX

	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:583)
	at org.openqa.selenium.remote.RemoteWebDriver$RemoteTargetLocator.defaultContent(RemoteWebDriver.java:1042)
	at tests.ExampleTest.testFrames(ExampleTest.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1215)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.run(TestNG.java:1048)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
Caused by: java.lang.NullPointerException
	at org.openqa.selenium.remote.TracedCommandExecutor.execute(TracedCommandExecutor.java:47)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:559)
```

**Comment by diemol** [comment]
This is a duplicate of #10116, we have already a fix that will be released in either on 15.12 or 16.12.

**Comment by MihirModiAC** [comment]
> This is a duplicate of #10116, we have already a fix that will be released in either on 15.12 or 16.12.

Any idea when this will be released? Thank you.

**Comment by stevenerat** [comment]
4.1.1 was just released and I believe the change log entry “BugFix: Handle null parameters in TracedCommandExecutor” refers to this fix.
